src/core/utils/fs.cpp

```diff
@@ -439,18 +439,25 @@ QString Utils::Fs::QDesktopServicesCacheLocation()
 
 QString Utils::Fs::QDesktopServicesDownloadLocation()
 {
-#if defined(Q_OS_WIN) || defined(Q_OS_OS2)
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
+    return QStandardPaths::writableLocation(QStandardPaths::DownloadLocation);
+#else // Qt4
+
+#if defined(Q_OS_OS2)
     // as long as it stays WinXP like we do the same on OS/2
     // TODO: Use IKnownFolderManager to get path of FOLDERID_Downloads
     // instead of hardcoding "Downloads"
     // Unfortunately, this would break compatibility with WinXP
-#if (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
     return QDir(QDesktopServices::storageLocation(QDesktopServices::DocumentsLocation)).absoluteFilePath(
                 QCoreApplication::translate("fsutils", "Downloads"));
-#else
-    return QDir(QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation)).absoluteFilePath(
-                QCoreApplication::translate("fsutils", "Downloads"));
 #endif
+
+#if defined(Q_OS_WIN)
+    if (QSysInfo::windowsVersion() <= QSysInfo::WV_XP)  // Windows XP
```

Comment:
This needs to be done for qt5+win XP too.
I guess this is the final change. So please update your v3_2_x PR too.